File: rxhttp/src/main/java/com/senierr/http/internal/MethodBuilder.java
package com.senierr.http.internal;

import android.support.annotation.NonNull;

/**
 * Http请求方法构建器
 *
 * @author zhouchunjie
 * @date 2018/8/29
 */
public class MethodBuilder {

    private @NonNull String value;

    private MethodBuilder(@NonNull String value) {
        this.value = value;
    }

    public static MethodBuilder get() {
        return new MethodBuilder("GET");
    }

    public static MethodBuilder post() {
        return new MethodBuilder("POST");
    }

    public static MethodBuilder put() {
        return new MethodBuilder("PUT");
    }

    public static MethodBuilder delete() {
        return new MethodBuilder("DELETE");
    }

    public static MethodBuilder head() {
        return new MethodBuilder("HEAD");
    }

    public static MethodBuilder patch() {
        return new MethodBuilder("PATCH");
    }

    public static MethodBuilder options() {
        return new MethodBuilder("OPTIONS");
    }

    public static MethodBuilder trace() {
        return new MethodBuilder("TRACE");
    }

    public String build() {
        return value;
    }
}

File: rxhttp/src/main/java/com/senierr/http/builder/MethodBuilder.kt
package com.senierr.http.builder

/**
 * Http请求方法构建器
 *
 * @author zhouchunjie
 * @date 2018/8/29
 */
class MethodBuilder(private val value: String) : Builder<String> {

    companion object {
        const val GET = "GET"
        const val POST = "POST"
        const val PUT = "PUT"
        const val DELETE = "DELETE"
        const val HEAD = "HEAD"
        const val PATCH = "PATCH"
        const val OPTIONS = "OPTIONS"
        const val TRACE = "TRACE"
    }

    override fun build(): String {
        return value
    }
}